### PR TITLE
Refine incident page

### LINKED ISSUE #2211: Misleading documentation on Incidents for unexpected exceptions
[Documentation on incidents](https://docs.camunda.io/docs/components/concepts/incidents/) states that "Incidents are not created when an unexpected exception (e.g. NullPointerException, OutOfMemoyError etc.) occurs."

When there is a nullPointerException thrown in the worker, an incident does get created. This information is misleading. If this is documented from zeebe's perspective, where a process instance gets blacklisted and doesn't show up in Operate, then this has to be clarified in the documentation, else it is confusing the users.

### PR DESCRIPTION
**Pull request by akeller**
## Description

Closes #2211.

## When should this change go live?

- [ ] This change is not yet live and should not be merged until {ADD_DATE} (apply `hold` label or convert to draft PR)?
- [x] There is no urgency with this change.
- [ ] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [ ] This functionality is already available but undocumented.
- [x] This is a bug fix or security concern.

## PR Checklist

- [x] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [x] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

**Review by jothikiruthika — APPROVED** [review]
I am assuming that we don't want to mention the blacklisting concept here. Looks ok. Not misleading anymore.

File: docs/components/concepts/incidents.md [hunk 1]
```diff
@@ -6,15 +6,15 @@ description: "A process instance is stuck at a particular point, and requires us
...
+Incidents are created in the following scenarios:
 
-- A job is failed and it has no retries left.
-- An input or output variable mapping can't be applied.
-- A condition can't be evaluated.
-- A decision can't be evaluated.
+- Explicitly by job workers sending FAIL commands
+- Implicitly when a job is out of retries
+- Input/output-mapping cannot be applied
+- A condition or decision cannot be evaluated
```
Comment (oleschoenburg) [inline docs/components/concepts/incidents.md:14]: @akeller Sorry to go back on this but could we keep the original text here and in the versioned docs for 8.2? It was more precise, for example " Explicitly by job workers sending FAIL commands" (which I suggested) is not correct, it only creates an incident when no retries are left.

**Review by oleschoenburg — COMMENTED** [review]
Thanks for jumping on it @akeller. The main change regarding unexpected error looks good :+1:

File: docs/components/concepts/incidents.md (hunk 1 above)
Comment (akeller) [reply docs/components/concepts/incidents.md:14]: I will adjust.

**Comment by akeller** [comment]
@oleschoenburg can I get a review again? This PR only includes the admonition language now.

**Review by oleschoenburg — APPROVED** [review]
Sorry, I'm not used to this repo's policy of discarding stale reviews :sweat_smile: 

Thanks again, looks good :+1:

**Pull request merged**